testing/sdk_utils.py

```diff
@@ -88,37 +89,57 @@ def dcos_version_at_least(version):
 
 
 def check_dcos_min_version_mark(item: pytest.Item):
-    '''Enforces the dcos_min_version pytest annotation, which should be used like this:
+    """Enforces the dcos_min_version pytest annotation, which should be used like this:
 
     @pytest.mark.dcos_min_version('1.10')
     def your_test_here(): ...
 
     In order for this annotation to take effect, this function must be called by a pytest_runtest_setup() hook.
-    '''
-    min_version_mark = item.get_marker('dcos_min_version')
+    """
+    min_version_mark = item.get_marker("dcos_min_version")
     if min_version_mark:
         min_version = min_version_mark.args[0]
-        message = 'Feature only supported in DC/OS {} and up'.format(min_version)
-        if 'reason' in min_version_mark.kwargs:
-            message += ': {}'.format(min_version_mark.kwargs['reason'])
+        message = "Feature only supported in DC/OS {} and up".format(min_version)
+        if "reason" in min_version_mark.kwargs:
+            message += ": {}".format(min_version_mark.kwargs["reason"])
         if dcos_version_less_than(min_version):
             pytest.skip(message)
 
 
 def is_open_dcos():
-    '''Determine if the tests are being run against open DC/OS. This is presently done by
-    checking the envvar DCOS_ENTERPRISE.'''
-    return not (os.environ.get('DCOS_ENTERPRISE', 'true').lower() == 'true')
+    """Determine if the tests are being run against open DC/OS. This is presently done by
+    checking the envvar DCOS_ENTERPRISE."""
+    return not (os.environ.get("DCOS_ENTERPRISE", "true").lower() == "true")
 
 
 def is_strict_mode():
-    '''Determine if the tests are being run on a strict mode cluster.'''
-    return os.environ.get('SECURITY', '') == 'strict'
+    """Determine if the tests are being run on a strict mode cluster."""
+    return os.environ.get("SECURITY", "") == "strict"
+
+
+def is_cli_supports_service_version_upgrade():
```

Comment:
TBH I'd just put these checks for CLI support in `sdk_upgrade.py` for now. They're specific to the upgrade/update operation itself, and ideally nobody outside of `sdk_upgrade` should need to know about the different variants, they'd just be handled by `sdk_upgrade` internally.